apps/docs/docs/dev/12-jayvee-testing.md

```diff
@@ -0,0 +1,247 @@
+- Language-server constraint validation (located [here](https://github.com/jvalue/jayvee/tree/dev/libs/language-server/src/lib/constraint))
+- Custom block (property) validation of the three existing extensions (std extension located [here](https://github.com/jvalue/jayvee/blob/dev/libs/extensions/std/lang/src))
+- Grammar validation tests for all official full examples from the [/example](https://github.com/jvalue/jayvee/tree/main/example) folder (located [here](https://github.com/jvalue/jayvee/blob/dev/libs/extensions/std/lang/src/example-validation.spec.ts))
+- Grammar validation tests for all block examples of the std extension (located [here](https://github.com/jvalue/jayvee/blob/dev/libs/extensions/std/lang/src/meta-inf-example-validation.spec.ts))
+
+## Execution tests
+These kind of tests are mainly located inside the [interpreter](https://github.com/jvalue/jayvee/tree/main/libs/language-server), its [interpreter-lib](https://github.com/jvalue/jayvee/tree/dev/libs/interpreter-lib), the [execution lib](https://github.com/jvalue/jayvee/tree/dev/libs/execution) as well as the execution parts of each extension (for example [std/exec](https://github.com/jvalue/jayvee/tree/main/libs/extensions/std/exec)).
+
+### Testing utils
+The testing utils for execution tests are spread between the extensions, with the interfaces and base utils located inside the [execution lib](https://github.com/jvalue/jayvee/tree/dev/libs/execution).  
+They can be imported using `@jvalue/jayvee-extensions/rdbms/test`, `@jvalue/jayvee-extensions/std/test` and `@jvalue/jayvee-execution/test`.
+
+[**utils.ts**](https://github.com/jvalue/jayvee/blob/dev/libs/execution/test/utils.ts):  
+At the moment this only contains two functions: `clearBlockExecutorRegistry` and `clearConstraintExecutorRegistry`. These are used to clear the corresponding registries between tests, in case the tested method initializes Jayvee itself (see [smoke test](#existing-tests-1)).
+
+[**block-executor-mocks.ts**](https://github.com/jvalue/jayvee/blob/dev/libs/execution/test/block-executor-mock.ts):  
+`BlockExecutorMock` interface for defining mocks for `AbstractBlockExecutor`. Generally only loader and executor blocks require mocks, because they interact with "the outside world" (i.e. `HttpExtractor` making http calls).  
+Due to how vastly different each `BlockExecutor` can be, this interface is very simple, containing only a `setup(...args: unknown[])` and a `restore()` method. See below for existing implementations.
+
+[**rdbms/exec/test**](https://github.com/jvalue/jayvee/tree/dev/libs/extensions/rdbms/exec/test):  
+Contains the implementation of `BlockExecutorMock` for `PostgresLoaderExecutor` and `SQLiteLoaderExecutor`.  
+Both of these executors are mocked using `jest.mock` to mock the corresponding libraries (`pg` and `sqlite3`) (**Note:** This part has to be added to the top of each `*.spec.ts` file (see usage below) due to how `jest.mock` works).  
```

Comment:
Rather make it a comment in the code.

Reply:
I am guessing you mean
> **Note:** This part has to be added to the top of each `*.spec.ts` file (see usage below) due to how `jest.mock` works

and that I should add this in the code example below?

Reply:
If I am right, I choose to put it here, because I honestly could not come up with a wording that would not sound completely strange as a comment

Reply:
Sth like this?
```
// Global mocking of external library required, even though the mocking is encapsulated in helper classes
```